src/Microsoft.Azure.WebJobs.Host/Queues/DispatchProcessor.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Queues
+{
+    /// <summary>
+    /// wrapper around shared queue
+    /// </summary>
+    public class DispatchProcessor
```

Comment:
For the types/concepts that are exposed via ListenerFactoryContext to users, I think we should consider better names. Rather than DispatchProcessor/DispatchProcessorFactory, how about MessageQueue/MessageQueueFactory? In the end, from the user's perspective, it really is a general message queue.